### ISSUE TITLE
stop asking me to install the nuget provider

### ISSUE THREAD

**Issue by madewokherd** [OneGet/oneget#153]
Whenever I do a find-package, I get this message:

The provider 'nuget v2.8.5.127' is not installed.
nuget may be manually downloaded from https://oneget.org/nuget-anycpu-2.8.5.127.exe and installed.
Would you like PackageManagement to automatically download and install 'nuget' now?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"):

There should be an option to never ask again, and/or it should explain why the nuget provider is needed.

**Comment by dsolodow** [comment]
If you don't want it to ask, you'd either need to install the provider (either by saying yes or by using install-packageprovider) or by disabling the chocolately and nuget providers.

**Comment by madewokherd** [comment]
I don't have the chocolatey provider installed.

If disabling is what I have to do to not get that prompt again, then the prompt should offer to do that.

**Comment by Jaykul** [comment]
I agree with @madewokherd -- that prompt should tell us what provider is asking for the nuget provider to be installed.

In general, inter-provider dependencies should be exposed in some way, and when a provider's trying to install another provider, the information about the calling provider should be in the install prompt.

Currently, is there even any indication (or documentation?) that such dependencies are even possible?

FWIW, currently the built-in implementations of both the PSModule and Chocolatey providers depend on the NuGet provider, so you'd have to remove (unregister) any package source that used the Chocolatey, NuGet, or PSModule providers... at least.

**Comment by madewokherd** [comment]
I have the PSModule provider, but I'm not sure why. Is that installed by default?

**Comment by Jaykul** [comment]
Yeah, PSModule is the PowerShellGet module -- it's the real reason OneGet was promoted to in-box so quickly ;-)

**Comment by madewokherd** [comment]
It's not really in-box if it has an external dependency, is it?

**Comment by jianyunt** [comment]
The message has been fixed in the latest WMF 5.0 as below: 

PowerShellGet requires NuGet provider version '{0}' or newer to interact with NuGet-based repositories. The NuGet provider must be available in '{1}' or '{2}'. You can also install the NuGet provider by running 'Install-PackageProvider -Name NuGet -MinimumVersion {0} -Force'. Do you want PowerShellGet to install and import the NuGet provider now?

But I'd like to keep this issue open because PowerShellGet still needs to import-pp after the install-pp to avoid further prompt.

**Comment by tillig** [comment]
I just ran into this and found that even though I had `2.8.5.205` in _both_ the main `Program Files` folder _and_ in my user folder, I was still prompted for "version `2.8.5.201` or newer". The only way to get the prompt to stop was by installing _exactly_ version `2.8.5.201`.

**Comment by jianyunt** [comment]
@tillig are you using TP4 or TP5?
can you check if there are any old version of NuGet provider under 

 $env:ProgramFiles\PackageManagement\ProviderAssemblies\
$env:LOCALAPPDATA\PackageManagement\ProviderAssemblies\

**Comment by tillig** [comment]
I recently upgraded from PowerShell 4 to PowerShell 5 on Windows Server 2012R2. Apparently that's not a clean thing.

I deleted everything in both of the folders you mentioned and re-ran the `Install-PackageProvider` command and _now_ things are working without prompting me. I can't tell what's different - it appears the folder structure and everything are the same, but it's working again. Thanks for the pointer.
